crates/scroll/evm/src/config.rs

```diff
@@ -161,6 +212,52 @@ impl ConfigureEvmEnv for ScrollEvmConfig {
     }
 }
 
+pub(crate) trait ScrollConfigureEvm: ConfigureEvm {
```

Comment:
Because of the new GAT on `ConfigureEvm` which shadows the provided configured Evm and just bounds it to `Evm<'a, DB, I>`, this trait is introduced to provide an alternative Evm which is additionally bound by `ScrollEvmT`, providing access to internal state such as `l1_fee` and `base_fee`.

Reply:
Isn't it sufficient that we implement `ScrollEvmT` on `ScrollEvm` and then in `ConfigureEvm` we have `Evm = ScrollEvm`.  Then when we use it in `executre.rs:65` we have `ConfigureEvm<Header = N::BlockHeader, Transaction = N::SignedTx, Evm = ScrollEvm>`. This would eliminate the need for the additional requirement for `ScrollConfigureEvm`. I'm not sure about this and may not be justified to look into but I'm curious.

Reply:
I did try to go down that way at the start but hit an issue because the `Evm` GAT has the following generics: `Evm<'a, DB + 'a, I + 'a>`. This means that the concrete type on `Evm` needs to be written with: `ConfigureEvm<Header = N::BlockHeader, Transaction = N::SignedTx, Evm<'a, DB, ()> = ScrollEvm<'a, DB, ()>>`. The `I` bound can be the unit type, since the `evm_for_block` function returns unit for the inspector, but the `'a` lifetime must come from somewhere, and must also bound the `DB` generic because of `Evm<'a, DB + 'a, I + 'a>`. I can add this as phantom on `ScrollExecutionStrategy` and `ScrollExecutionStrategyFactory` if you think this solution is preferable, I was trying to avoid having phantom markers.

Reply:
Thanks for the explanation! No this is fine.